DNN Platform/Library/UI/WebControls/PropertyEditor/ProfileEditorControl.cs

```diff
@@ -73,8 +73,9 @@ protected override void CreateEditor()
 
             foreach (FieldEditorControl editor in Fields)
             {
-                //Check whether Field is readonly
-                string fieldName = editor.Editor.Name;
+                //Check whether Field is readonly. Replace "_" with " " (Spaces) to find the correct definitions.
+                string fieldName = editor.Editor.Name.Replace("_", " ");
```

Comment:
I think that combined with the modified SP this is still not good @ricardodiasc. Consider this

1. User creates the property `Hire_date`
1. `editor.Editor.Name` will be the same
1. the code above will replace underscores with space so that `fieldName  == "Hire date"`
   * profile property will not be found

If we don't want to change much C# code the the only way to be fool proof is to ensure that the property added to the database does not include any underscores in the first place.

The stored procedure should not only match by the normalised name but also stored that normalised name in the table. Also leading and trailing whitespace would probably have to be removed if it's not done already (after underscores are replaced).

Please see my description in #2096